common/src/api/external/mod.rs

```diff
@@ -2124,6 +2145,376 @@ impl std::fmt::Display for Digest {
     }
 }
 
+/// An address lot and associated blocks resulting from creating an address lot.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct AddressLotCreateResponse {
+    /// The address lot that was created.
+    pub lot: AddressLot,
+
+    /// The address lot blocks that were created.
+    pub blocks: Vec<AddressLotBlock>,
+}
+
+/// Represents an address lot object, containing the id of the lot that can be
+/// used in other API calls.
+#[derive(
+    ObjectIdentity, Clone, Debug, Deserialize, JsonSchema, Serialize, PartialEq,
+)]
+pub struct AddressLot {
+    #[serde(flatten)]
+    pub identity: IdentityMetadata,
+}
```

Comment:
Looking back on the RFD, it seems like we're missing a few things here. (I know you wrote this, I'm just restating for my own benefit). It looks like there should be a [`kind`](https://rfd.shared.oxide.computer/rfd/0267#addresslotkind_enum) which can be `infra` or `pool`. Also the relationship to blocks is interesting... is `blocks` an unbounded collection? Is it possible to inline an array here or does it _have_ to be paginated?

Reply:
I feel like we at least want _some_ information here. Maybe it's the block count or a list of the first 20 or so blocks?

Reply:
Yeah the `kind` should be there for sure. Having the number of blocks would also be good. I think leaving the blocks as a paginated interface is probably the way to go. I can imagine someone carving up something like `10.0.0.0/8` into tens of thousands of blocks for a large deployment.

Reply:
@internet-diglett we still need to implement `kind` here.